### PR TITLE
JWT authentication

### PR DESCRIPTION
**Pull request by ebosetalee**
Here are the changes @imadu

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+JWT="our_crypto_secret234"
```
Comment (imadu) [inline .env]: You're not supposed to commit your env files. That's where all our secret stuff goes into. You should add it to your git ignore

File: controllers/login.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+const { createUser, getUserByEmail } = userController;
+
+const signUp = async (req, res) => {
+    try {
+        const { emailAddress, password, username, role } = req.body;
+        const user = await getUserByEmail(emailAddress);
+        if (user) {
+            throw Error("User Already Exists");
+        } else {
```
Comment (imadu) [inline controllers/login.js]: Remove the else block, we don't need it .

File: controllers/users.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
 import Users from "../models/users.js";
+import bcrypt from "bcrypt";
+
+async function hashPassword(password) {
```
Comment (imadu) [inline controllers/users.js]: Actually, this shouldn't be an async function. If I remember correctly, the `bcrypt.hash()` function doesn't return a promise, so it should be without the async/await keyword

**Review by imadu — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (ebosetalee) [reply .env]: my goodness i forgot

Comment (ebosetalee) [reply .env]: i need to reverse my commit

File: controllers/users.js (hunk 3 above)
Comment (ebosetalee) [reply controllers/users.js]: ok

File: controllers/login.js (hunk 2 above)
Comment (ebosetalee) [reply controllers/login.js]: ok

**Review by imadu — APPROVED** [review]

**Pull request merged**